catalog/resource_catalog_test.go

```diff
@@ -343,6 +343,68 @@ func TestCatalogCreateDeltaSharing(t *testing.T) {
 	}.ApplyNoError(t)
 }
 
+func TestCatalogCreateForeign(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   "POST",
+				Resource: "/api/2.1/unity-catalog/catalogs",
+				ExpectedRequest: catalog.CatalogInfo{
+					Name:    "foreign_catalog",
+					Comment: "b",
+					Options: map[string]string{
```

Comment:
Somehow, the request doesn't match the expected request here and does not include the "options" field, I am not sure if I am missing anything - but there doesn't seem to be anything extra needed to configure this.

Reply:
```
=== FAIL: catalog TestCatalogCreateForeign (0.14s)
    testing.go:427: 
        	Error Trace:	/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/qa/testing.go:427
        	            				/opt/hostedtoolcache/go/1.19.12/x64/src/net/http/server.go:2109
        	            				/opt/hostedtoolcache/go/1.19.12/x64/src/net/http/server.go:2947
        	            				/opt/hostedtoolcache/go/1.19.12/x64/src/net/http/server.go:1991
        	            				/opt/hostedtoolcache/go/1.19.12/x64/src/runtime/asm_amd64.s:1594
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"comment":"b", "connection_name":"foo", "name":"a", "options":map[string]interface {}{"a":"b"}}
        	            	actual  : map[string]interface {}{"comment":"b", "connection_name":"foo", "name":"a"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,8 +1,5 @@
        	            	-(map[string]interface {}) (len=4) {
        	            	+(map[string]interface {}) (len=3) {
        	            	  (string) (len=7) "comment": (string) (len=1) "b",
        	            	  (string) (len=15) "connection_name": (string) (len=3) "foo",
        	            	- (string) (len=4) "name": (string) (len=1) "a",
        	            	- (string) (len=7) "options": (map[string]interface {}) (len=1) {
        	            	-  (string) (len=1) "a": (string) (len=1) "b"
        	            	- }
        	            	+ (string) (len=4) "name": (string) (len=1) "a"
        	            	 }
        	Test:       	TestCatalogCreateForeign
        	Messages:   	json strings do not match
```

Reply:
cc @nkvuong @tanmay-db would appreciate some help here.

Reply:
the OpenAPI spec has been updated, but we need a new release of Go SDK to incorporate those changes - should have one shortly

Reply:
The newest version of Go SDK was released 2 days back and contains options field for Create and Update catalog.
![image](https://github.com/databricks/terraform-provider-databricks/assets/88379306/3e2797bc-eb2f-4a3e-985a-a59edd68fabb)

https://github.com/databricks/databricks-sdk-go/releases/tag/v0.18.0

Were these the changes that the PR is waiting for?

Reply:
I see, there are some issues with updating go sdk to latest version in terraform for now. The PR would need to be on hold until they are resolved. We are working on them.